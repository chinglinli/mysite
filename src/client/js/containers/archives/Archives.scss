@import '../../../css/base.scss';

.root {
  padding: 2em;
  border-bottom: 1px solid #eee;
  background: white;

  .header {
    font-size: 2em;
    font-weight: 700;
    color: rgb(26, 26, 26);;
    margin-bottom: 1em;
  }

  table {
    border-collapse: collapse;
    width: 100%;
    font-size: 1.2em;

    tr {
      background: #eee;

      &:nth-of-type(even) {
        background: #f9f9f9;
      }

      td {
        padding: 0.6em;
        border: 1px solid #ccc;
      }
    }

    .link {
      @extend .link;
    }
  }
}
