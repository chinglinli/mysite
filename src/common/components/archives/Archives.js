import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { Link } from 'react-router'
import dateformat from 'dateformat'

import styles from './Archives.scss'

export default class Archives extends React.Component {
  static PropTypes = {
    articles: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        slug: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        date: PropTypes.any.isRequired
      })
    )
  }

  render () {
    const {articles} = this.props
    if (articles.isEmpty()) {
      return <article>讀取中……</article>
    }

    return (
      <article className={styles.root}>
        <div>
          <header className={styles.header}>
            所有文章列表
          </header>
          <table className={styles.table}>
            <tbody>
            {articles.map(article => (
              <tr key={article.get('slug')}>
                <td>{dateformat(article.get('date'), 'yyyy/mm/dd')}</td>
                <td>
                  <Link className={styles.link} to={`/articles/${article.get('slug')}/`}>{article.get('title')}</Link>
                </td>
              </tr>
            ))}
            </tbody>
          </table>
        </div>
      </article>
    )
  }
}
