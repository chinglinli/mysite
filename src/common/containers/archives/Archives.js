import * as React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { connect } from 'react-redux'
import { Link } from 'react-router'

import * as articleActions from '../../ducks/article/actions'
import * as articleSelectors from '../../ducks/article/selectors'

import styles from './Archives.scss'

export class Archives extends React.Component {
  static PropTypes = {
    articles: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        slug: React.PropTypes.string,
        title: React.PropTypes.string,
        date: React.PropTypes.any.isRequired
      })
    ),
    fetchArticles: React.PropTypes.func
  }

  componentWillMount () {
    if (this.props.articles.isEmpty()) {
      this.props.fetchArticles()
    }
  }

  render () {
    const {articles} = this.props
    if (articles.isEmpty()) {
      return <article>讀取中……</article>
    }

    return (
      <article className={styles.root}>
        <div>
          <header className={styles.header}>
            所有文章列表
          </header>
          <table className={styles.table}>
            <tbody>
            {articles.map(article => (
              <tr key={article.get('slug')}>
                <td>{article.get('date').format('YYYY/MM/DD')}</td>
                <td>
                  <Link className={styles.link} to={`/articles/${article.get('slug')}/`}>{article.get('title')}</Link>
                </td>
              </tr>
            ))}
            </tbody>
          </table>
        </div>
      </article>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    articles: articleSelectors.getAllArticles(state, ownProps)
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchArticles: () => dispatch(articleActions.fetchArticles())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Archives)
