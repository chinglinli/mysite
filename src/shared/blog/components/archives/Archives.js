import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import Link from '../../../generic/components/Link/index'
import dateformat from 'dateformat'

import Loading from '../Loading/index'

import styles from './Archives.scss'

export default class Archives extends React.PureComponent {
  static PropTypes = {
    articles: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        slug: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        date: PropTypes.any.isRequired
      })
    )
  }

  render () {
    const {articles} = this.props
    if (articles.isEmpty()) {
      return (
        <section className={styles.archives}>
          <Loading/>
        </section>
      )
    }

    return (
      <section className={styles.archives}>
        <article>
          <header>
            <h1><Link to='/articles/archives/'>所有文章列表</Link></h1>
          </header>
          <table>
            <tbody>
            {articles.map(article => (
              <tr key={article.get('slug')}>
                <td>{dateformat(article.get('date'), 'yyyy/mm/dd')}</td>
                <td>
                  <Link to={`/articles/${article.get('slug')}/`}>{article.get('title')}</Link>
                </td>
              </tr>
            ))}
            </tbody>
          </table>
        </article>
      </section>
    )
  }
}
