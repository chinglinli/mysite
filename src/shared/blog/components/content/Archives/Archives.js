import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import NormalLink from '../../generic/NormalLink'
import TitleLink from '../../generic/TitleLink'
import dateformat from 'dateformat'
import styled from 'styled-components'
import Loading from '../../generic/Loading'


const Base = styled.section`
  float: left;
  width: 800px;

  @media (max-width: 1200px) {
    width: 100%;
  }

  article {
    padding: 2em;
    border-bottom: 1px solid #eee;
    background: white;

    table {
      
    }
  }
`

const Header = styled.header`
  h1 {
    margin: 3px 0 24px;
  }
`

const ArticleTable = styled.table`
  border-collapse: collapse;
  width: 100%;
  font-size: 1.2em;

  tr {
    background: #eee;

    &:nth-of-type(even) {
      background: #f9f9f9;
    }

    td {
      padding: 0.6em;
      border: 1px solid #ccc;
    }
  }
`

export default class Archives extends React.PureComponent {
  static PropTypes = {
    articles: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        slug: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        date: PropTypes.any.isRequired
      })
    )
  }

  renderHeader = () => {
    return (
      <Header>
        <h1><TitleLink to='/articles/archives/'>所有文章列表</TitleLink></h1>
      </Header>
    )
  }

  renderArticleTable = () => {
    const {articles} = this.props
    return (
      <ArticleTable>
        <tbody>
        {articles.map(article => (
          <tr key={article.get('slug')}>
            <td>{dateformat(article.get('date'), 'yyyy/mm/dd')}</td>
            <td><NormalLink to={`/articles/${article.get('slug')}/`}>{article.get('title')}</NormalLink></td>
          </tr>
        ))}
        </tbody>
      </ArticleTable>
    )
  }

  render () {
    const {articles} = this.props
    if (articles.isEmpty()) {
      return (
        <Base>
          <Loading/>
        </Base>
      )
    }

    return (
      <Base>
        <article>
          {this.renderHeader()}
          {this.renderArticleTable()}
        </article>
      </Base>
    )
  }
}
