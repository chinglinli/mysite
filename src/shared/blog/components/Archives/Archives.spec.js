import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Archives from './Archives'

test('It should render Archives component with empty articles correctly', () => {
  const articles = Immutable.fromJS([])

  const component = renderer.create(
    <Archives articles={articles}/>
  )
  expect(component.toJSON()).toMatchSnapshot()
})

test('It should render Archives component correctly', () => {
  const articles = Immutable.fromJS([
    {slug: 'slug 1', title: 'title 1', date: new Date(2017, 4, 15)},
    {slug: 'slug 2', title: 'title 2', date: new Date(2017, 4, 16)},
    {slug: 'slug 3', title: 'title 3', date: new Date(2017, 4, 17)}
  ])

  const component = renderer.create(
    <Archives articles={articles}/>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
