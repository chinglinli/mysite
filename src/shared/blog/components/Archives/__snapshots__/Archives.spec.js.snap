// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Archives component correctly 1`] = `
<section
  className="archives"
>
  <article>
    <header>
      <h1>
        <a
          onClick={[Function]}
          style={Object {}}
        >
          所有文章列表
        </a>
      </h1>
    </header>
    <table>
      <tbody>
        <tr>
          <td>
            2017/05/15
          </td>
          <td>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              title 1
            </a>
          </td>
        </tr>
        <tr>
          <td>
            2017/05/16
          </td>
          <td>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              title 2
            </a>
          </td>
        </tr>
        <tr>
          <td>
            2017/05/17
          </td>
          <td>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              title 3
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </article>
</section>
`;

exports[`It should render Archives component with empty articles correctly 1`] = `
<section
  className="archives"
>
  <div
    className="Loading__Base-s1d02ruw-0 idafHP"
  >
    <div
      className="Loading__LoadingImage-s1d02ruw-1 cjrynz"
    />
    <div
      className="Loading__Message-s1d02ruw-2 fgoCqC"
    >
      正在努力讀取中…
    </div>
  </div>
</section>
`;
